src/StoreKeeper/WooCommerce/B2C/Cron/ProcessTaskCron.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\Cron;
+
+use Exception;
+use StoreKeeper\WooCommerce\B2C\Commands\ProcessAllTasks;
+use StoreKeeper\WooCommerce\B2C\Core;
+use StoreKeeper\WooCommerce\B2C\Exceptions\BaseException;
+use StoreKeeper\WooCommerce\B2C\Exceptions\InvalidRunnerException;
+use StoreKeeper\WooCommerce\B2C\Options\CronOptions;
+use Throwable;
+
+class ProcessTaskCron
+{
+    public const TASK_LIMIT = 100;
+
+    /**
+     * @throws BaseException
+     * @throws Exception|Throwable
+     */
+    public function execute(): void
+    {
+        $beforeCount = ProcessAllTasks::countNewTasks();
+        try {
+            CronRegistrar::validateRunner(CronRegistrar::RUNNER_WPCRON);
+            CronOptions::set(CronOptions::LAST_PRE_EXECUTION_DATE, date(DATE_RFC2822));
+            $runner = Core::getCommandRunner();
+            $runner->execute(ProcessAllTasks::getCommandName(), [], [
+                'limit' => self::TASK_LIMIT,
+            ]);
+            CronOptions::updateSuccessfulExecution();
+        } catch (InvalidRunnerException $invalidRunnerException) {
+            $beforeCount = 0;
+            throw $invalidRunnerException;
+        } catch (Throwable $throwable) {
+            CronOptions::updateFailedExecution($throwable);
+            throw $throwable;
+        } finally {
+            CronOptions::updateHasProcessed($beforeCount);
+        }
+    }
```

Comment:
Maybe make a function out of it? cos coppied few times

```suggestion
    public function execute(): void
    {
        $this->withCronCheck( CronRegistrar::RUNNER_WPCRON, function() {
            $runner = Core::getCommandRunner();
            $runner->execute(ProcessAllTasks::getCommandName(), [], [
                'limit' => self::TASK_LIMIT,
            ]);
        }); 
    }
```

Reply:
Made a function, will callback, catch and invalid parameter too.